spark/dl/src/main/scala/com/intel/analytics/bigdl/models/utils/ModelBroadcast.scala

```diff
@@ -88,21 +95,39 @@ class ModelBroadcast[T: ClassTag](implicit ev: TensorNumeric[T]) extends Seriali
     weightsBias
   }
 
-  private def putWeightBias(broadcastWeightBias: Array[Tensor[T]],
-    localModel: Module[T]): Unit = {
-    val localWeightBias = localModel.parameters()._1
+  private def putWeightBias(
+        broadcastWeightBias: Array[Tensor[T]],
+        localModel: Module[T],
+        inference: Boolean = true): Unit = {
+    val (localWeightBias, localGradWeightBias) = localModel.parameters()
     var i = 0
     while (i < localWeightBias.length) {
       if (localWeightBias(i) != null) {
         localWeightBias(i).set(broadcastWeightBias(i))
       }
       i += 1
     }
+    // init gradient with a compacted storage
+    if (!inference) {
```

Comment:
I think you should move this logic to a separated function. It has nothing to do with the putWeightBias function